### DISCUSSION TITLE
awardachievement able to update unlock timestamp

### DISCUSSION THREAD

**Discussion by meleu** [RetroAchievements/RAWeb#800] [Ideas]
## feature request

Add an "update flag" to the request responsible to unlock an achievement, so players can update their "Achievement Unlocked" timestamps.

## problem to be solved

It's quite common to "force" the users to reset their progress in some games so they can be able to join a tournament (like leapfrog, AotW or even when they organize their own tournaments between friends).

A worse approach: sometimes they create alternative accounts just to join the tournament without messing with their "official" accounts.

Both "solutions" are not the ideal way to allow players to join tournaments.

If we have a simple flag to tell the server "hey, I unlocked this achievement again, please, update the timestamp." those actions would be unnecessary.

## to be done later

After this feature is implemented in RAWeb, we must add something like an "Update Timestamp" toggle in RAIntegration and RetroArch.

I would like to know your opinions @Jamiras, @luchaos, and `@everyone`.

**Comment by Xymjak** [comment]
I'll copypaste here my message from discord.

The database should have at least TWO columns - "the first time unlocked" which is NOT updating and "the last time unlocked"* which is updating - this would be a reasonable solution.
(The first used for masteries dates recalcs/badges/stats, the second used for events.)

I'm against updating an original timestamp since it's no different to resets, I still will use an alt for events. (and if updating an original timestamp will be forced in an event - I'll simply drop these events).
Adding "the last time unlocked"-column will allow me to drop my alt.

*can be as "all times unlocked"-column but not necessarily.

**Comment by Jamiras** [comment]
Another option would be to clone the achievement, sort of like we do with leaderboards of the month: 
![image](https://user-images.githubusercontent.com/32680403/133115661-2f901759-47c3-4f6a-9b94-7f526a208069.png)

We'd just have to have some way to include the cloned achievement in the list of achievements retrieved from the server for the game, without including it in the list of achievements required for master of the game. Probably by assigning it a new category (Core/Unofficial/**Event**).

See also my proposal for cloning achievements into an Event "game": https://github.com/RetroAchievements/RAWeb/discussions/665#discussioncomment-1127171

**Comment by Jamiras** [comment]
Event achievements (#2778) allow re-unlocking an achievement for an event without having to reset it.

Note that this only works when the event is looking at specific achievements. But if it's not looking at specific achievements, the player should be able to find an alternative achievement to unlock.
